contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+
+slot=()
+
+function show_usage ()
+{
+         echo "Usage: $0 [OPTIONS] [TARGET]"
+         echo " OPTIONS"
+         echo "  -t arg  Set number of parallel process; default 8"
+         echo "  -i arg  input FILE of table names; default: dump all classes"
+         echo "  -u arg  Set database user name; default dba"
+         echo "  -p arg  Set dbuser password"
+         echo "  -D arg  Set directory for unloaddb output dir/files"
+         echo "  -v      Set verbose mode on"
+
+         echo ""
+         echo " EXAMPLES"
+         echo "  $0 -t 4 -v demodb          # unload all tables in demodb"
+         echo "  $0 -i file demodb          # unload tables listed in file in demodb"
+         echo "  $0 -u user1 -D /tmp -i file -t 4 -v demodb"
+         echo ""
+}
+
+#
+# Kill unloaddb processes in progress and delete the directory, and delete incomplete files
+# Directories/files created by the normally terminated unloaddb process is not deleted.
+#
+function cleanup ()
+{
+        local i
+        local pid
+        local exit_stat
+
+	echo "interrupted"
+
+        for ((i = 0; i < $num_proc; i++))
+        do
+                if [ -f $database.$i/unloaddb_$i.pid ];then
+                        pid=$(cat $database.$i/unloaddb_$i.pid)
+
+                        kill -0 $pid 2> /dev/null
+                        if [ $? -eq 0 ];then
+                                kill -9 $pid
+                                rm -rf $database.$i
+                        fi
+                fi
+        done
+}
+
+function get_password ()
+{
+	local password
+
+	read -sp "Enter Password : " password
+	echo $password
+	echo "" > /dev/tty
+}
+
+function verify_user_pass ()
+{
+	local msg
+	local USERNAME
+	local dbuser
+	local passwd
+	local qry_dba_grp="SELECT u.name FROM db_user AS u, TABLE(u.groups) AS g(x) where x.name = 'DBA'"
+	local retcode
+
+	dbuser=$(echo ${user} | cut -d' ' -f2-2)
+	USERNAME=$(echo ${dbuser} | tr [:lower:] [:upper:])
+
+	# check the database server is running
+	retcode=$(ps -ef | grep cub_server | grep $database | wc -l)
+
+	if [ $retcode -eq 0 ];then
+		echo "Database server '$database' is not running"
+		exit 1
+	fi
+
+	dba_groups=$(csql -u public -l -c "$qry_dba_grp" $database | grep -w $USERNAME | wc -l)
+
+	if [ $USERNAME != "DBA" ] && [ $dba_groups -eq 0 ];then
+		echo "User '$dbuser' is not a member of DBA group"
+		exit 2
+	fi
+
+	# Try with NULL password
+	passwd=$(csql $user --password="" -c "SELECT 1" $database 2> /dev/null)
+
+	if [ $? -ne 0 ];then
+		passwd=$(get_password)
+		pass="-p $passwd"
+
+		passwd=$(csql $user $pass -c "SELECT 1" $database 2> /dev/null)
+		if [ $? -ne 0 ];then
+			echo "$dbuser: Incorrect or missing password"
+			exit 4
+		fi
+	fi
+}
+
+function get_options ()
+{
+         while getopts ":D:u:i:t:v" opt; do
+                case $opt in
+                        u ) user="-u $OPTARG" ;;
+                        i ) from_file="$OPTARG" ;;
+                        t ) num_proc="$OPTARG" ;;
```

Comment:
Is it possible to run more than 8 process ?
if it can't be executed and num_proc greater than 8, I think that you have to tune num_proc to 8.

Reply:
I set default num proc to 8, and max num proc to 16. 
And if a user set more than 16, for example '-t 17', it will be set to 16.
Thank you for your comment!